### PR TITLE
Use signatures in OpenQA::WebAPI::Plugin::REST

### PR DESCRIPTION
**Pull request by okurz**
After
* #4419

### CODE REVIEW

File: lib/OpenQA/WebAPI/Plugin/REST.pm [hunk 1]
```diff
@@ -2,49 +2,33 @@
...
-                Carp::croak "url is not a url"
-                  unless Scalar::Util::blessed $url && $url->isa('Mojo::URL');
-
-                return $self->tag('a', href => $url, 'data-method' => $method, @_);
-            }
-            else {
-                return '';
+            return '' unless $c->is_operator;
+            # Content
+            unless (ref $_[-1] eq 'CODE') {
```
Comment (perlpunk) [inline lib/OpenQA/WebAPI/Plugin/REST.pm:17]: there is still a reference to `@_`

Comment (perlpunk) [reply lib/OpenQA/WebAPI/Plugin/REST.pm:17]:
```suggestion
            unless (ref $args[-1] eq 'CODE') {
```

File: lib/OpenQA/WebAPI/Plugin/REST.pm [hunk 2]
```diff
@@ -2,49 +2,33 @@
...
-
-                return $self->tag('a', href => $url, 'data-method' => $method, @_);
-            }
-            else {
-                return '';
+            return '' unless $c->is_operator;
+            # Content
+            unless (ref $args[-1] eq 'CODE') {
+                $url = shift @args;
+                push @args, $content;
```
Comment (okurz) [inline lib/OpenQA/WebAPI/Plugin/REST.pm:19]: this code seems to be uncovered. Anyone has an idea what it could be useful for? Or should we just remove it?

Comment (Martchus) [reply lib/OpenQA/WebAPI/Plugin/REST.pm:19]: Not sure. So Git blame doesn't reveal anything?

Comment (okurz) [reply lib/OpenQA/WebAPI/Plugin/REST.pm:19]: Well, @aaannz has introduced it, we never changed it and it's not called in tests. What behavior could this be for?

Comment (aaannz) [reply lib/OpenQA/WebAPI/Plugin/REST.pm:19]: You mean @ancorgs ( commit https://github.com/os-autoinst/openQA/commit/34d51fb0c3c20d6a6e76bf13f5ae7e6ae52bd919 ). I have no idea what it is of and if it is still valid.

Comment (okurz) [reply lib/OpenQA/WebAPI/Plugin/REST.pm:19]: Well, I meant the refactoring in 04c265d57 but looks like you found a predecessor. I guess I will just remove that code.

**Review by kalikiana — APPROVED** [review]

**Review by Martchus — APPROVED** [review]

**Pull request merged**